website/content/docs/concepts/architecture.mdx

```diff
@@ -162,14 +176,129 @@ on hardware features such as architecture and availability of GPUs, or software
 like operating system and kernel version, or they can be business constraints like
 ensuring PCI compliant workloads run on appropriate servers.
 
+# Client Organization
+
+When the Nomad [scheduler][] receives a job registration request, it needs to
+determine which clients will run allocations for the job. By default, all nodes
+are considered for placements but this process can be adjusted by cluster
+operators and job submitters to achieve more control over where allocations
+are placed.
+
+The steps to achieve this control consists of setting certain values in
+clients' configuration and matching them in jobs, and so it's helpful to
+consider how clients will be configured based on your expected job registration
+patterns when planning your cluster deployment.
+
+There are several options that can be used depending on the desired outcome.
+
+### Affinities and Constraints
+
+Affinities and constraints allow users to define soft or hard requirements for
+their jobs. The [`affinity`][job_affinity] block specifies a soft requirement
+on certain node properties, meaning allocations for the job have a preference
+for some nodes, but may be placed elsewhere if the rules can't be matched,
+while the [`constraint`][job_constraint] block creates hard requirements and
+allocations can only be placed in nodes that match these rules. Job placement
+fails if a constraint cannot be satisfied.
+
+These rules can reference intrinsic node characteristics, which are called
+[node attributes][] and are automatically detected by Nomad, static values
+defined in the agent configuration file by cluster administrators, or dynamic
+values defined after the agent starts.
+
+Using affinities and constraints has the downside of only allowing allocations
+to gravitate towards certain nodes, but it does not prevent placements of jobs
+where the rules don't apply.
```

Comment:
```suggestion
Using affinities and constraints has the downside of only allowing allocations
to gravitate towards certain nodes, but it does not prevent placements of other
jobs for which the rules don't apply.
```

I think?

Reply:
Im a little confused here: constraints are hard requirements and "Job placement fails if a constraint cannot be satisfied." but at the same time they don't " it does not prevent placements of jobs where the rules don't apply" ?

Reply:
> I think?

Yup, that's right. Thanks!

> Im a little confused here: constraints are hard requirements and "Job placement fails if a constraint cannot be satisfied." but at the same time " it does not prevent placements of jobs where the rules don't apply" ?

Yeah, it's kind of weird to describe in text, so maybe I should put an example.

But imagine you have two jobs, one with a constraint and one without any constraints:

```hcl
job "app"  {
  constraint {
    attribute = "${meta.env}"
    value      = "prod"
  }
  # ...
}
```

```hcl
job "db"  {
  # ...
}
```

The `app` job is only allowed to run in clients with metadata `env=prod`, but the `db` is free to run anywhere, including clients with `env=prod`. 

So if you want to reserve the `env=prod` clients to only run `app` then you need a negative constraint in _all other_ jobs:

```hcl
job "db"  {
  constraint {
    attribute = "${meta.env}"
    operator  = "!="
    value      = "prod"
  }
  # ...
}
```

So constraints are kind of unidirectional in this sense: you can restrict a _job_ to specific nodes, but you can't restrict a _node_ to specific jobs.

Does this help?

Reply:
Updated this paragraph as I moved this section to the `Scheduler` section
https://github.com/hashicorp/nomad/blob/4f4ba9fdfc71eab2665c6091b249b711c6eb1f81/website/content/docs/concepts/scheduling/placement.mdx#L36-L38

Is this better or should I add an example to help illustrate the problem?